base_sms_client/models/sms_sms.py

```diff
@@ -69,7 +69,8 @@ def _check_gateway_method(self):
     @api.multi
     def _check_gateway_permission(self):
         self.ensure_one()
-        if self.gateway_id._check_permissions():
+        user = self.create_uid
+        if self.gateway_id._check_permissions(user):
```

Comment:
changing the method param can break compatibility in other modules (OCA or custom). I propose to fix this like:

```suggestion
        if self.sudo(self.create_uid.id).gateway_id._check_permissions():
```